fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/model/SophisticatedAdhocTransmissionModel.java

```diff
@@ -191,20 +208,19 @@ private Tuple<String, TransmissionResult> forwarding(
                     currentEntity.getPosition(),
                     currentEntity.getRadius()
             );
-            // get all reachable entities within singlehop range and remove sender
-            Map<String, SimulationNode> reachableEntities =
-                    TransmissionSimulator.getEntitiesInArea(currentNodes, singleHopReach);
-            reachableEntities.remove(senderName);
-            // try to find entity to build "pipeline" to destination area
-            String forwardingEntityName = getForwardingEntity(reachableEntities, receivers);
-            if (forwardingEntityName == null) { // if no entity to forward the message to was found, Forwarding fails
+            // get all reachable entities within singlehop range and remove current sender
+            final Map<String, SimulationNode> reachableEntities = TransmissionSimulator.getEntitiesInArea(currentNodes, singleHopReach);
+
+            // try to find next entity to build forwarding-chain towards destination area
+            final String forwardingEntityName = getForwardingEntity(reachableEntities, receivers);
+            if (forwardingEntityName == null || forwardingEntityName.equals(currentEntityName)) {
```

Comment:
I would leave this as-is for fail-safety, but I suspect that `forwardingEntityName == null` should never trigger, because as we included the sender now, it should always return _something_